### ISSUE TITLE
deepFreeze raises error on Buffer fields

### ISSUE THREAD

**Issue by astur** [snovakovic/js-flock#36]
When running smth like this:
`require('js-flock/deepFreeze')({b: Buffer.from('TEST')})`
it happens this error:
`TypeError: Cannot freeze array buffer views with elements`

**Comment by astur** [comment]
Same bug with `Int8Array` that isn't empty

**Comment by snovakovic** [comment]
@astur this is know issue and it's related to Object.freeze throwing that error when trying to freeze Buffer. 

Consider this code 

```javascript
var typedArray = new Uint32Array(4);
typedArray[0] = 10;

Object.freeze(typedArray); 
// throws TypeError : Cannot freeze array buffer views with elements(...)
```

deepFreeze is recursively applying Object.freeze so there is not much that can be done there. 

You can also reference this [stack overflow question](https://stackoverflow.com/questions/39141450/object-freeze-on-typedarray-and-arraybuffers-node-chrome-not-working-as-ex)

One line of thoughts was to skip freezing Buffers and avoid that exception for object that consist buffer. I decided not to do it for now as it would just silently ignore issue and posibly cause more problems then just throwing error. In you example from above 
`require('js-flock/deepFreeze')({b: Buffer.from('TEST')})`

if we ignore buffers from deepFreeze there won't be error but b would be unmodified which is not what the intention of the code was. So it's probably better to let user know that he can't do that rather then silently ignore operation.

Maybe workflow could be improved by adding additional option property etc skipBuffers and when set to true it would skip buffers. That way at least intention is clear. We can also enhance error message to something as 'Cannot freeze array buffer views with elements. You can skip freezing buffers with skipBuffer option'

@astur I'm interested to here your opinion on above matter?

**Comment by astur** [comment]
My opinion... well:

If you do smth like this:
`Object.freeze({b: Buffer.from('TEST')})`
everything will be ok. No any error.

I think it MUST be error on this:
`require('js-flock/deepFreeze')(Buffer.from('TEST'))`
but NOT on this:
`require('js-flock/deepFreeze')({b: Buffer.from('TEST')})`

`deepFreeze` applies `Object.freeze` recursively on every "object" field using `typeof prop === 'object'`, but it's true that `typeof Buffer.from('TEST') === 'object'`.

My opinion is: function deepFreeze must not doing recursion on Buffer.

**Comment by snovakovic** [comment]
@astur Thank you for your input!

Fixed in v3.0.2
